sdk/storage/azure-storage-blob/azure/storage/blob/__init__.py

```diff
@@ -81,3 +85,73 @@
     'ResourceTypes',
     'AccountPermissions',
 ]
+
+
+def upload_blob_to_url(
+        blob_url,  # type: str
+        data,  # type: Union[Iterable[AnyStr], IO[AnyStr]]
+        overwrite=False,  # type: bool
+        max_connections=1,  # type: int
+        encoding='UTF-8', # type: str
+        credential=None,  # type: Any
+        **kwargs):
+    """Upload data to a given URL
+
+    The data will be uploaded as a block blob.
+
+    :param str blob_url:
+        The full URI to the blob. This can also include a SAS token.
+    :param data:
+        The data to upload. This can be bytes, text, an iterable or a file-like object.
+    :type data: bytes or str or Iterable
+    :param bool overwrite:
+        Whether the blob to be uploaded should overwrite the current data.
+        If True, upload_blob_to_url will silently overwrite the existing data. If set to False, the
+        operation will fail with a ResourceExistsError.
+    :param credential:
+        The credentials with which to authenticate. This is optional if the
+        blob URL already has a SAS token. The value can be a SAS token string, an account
+        shared access key, or an instance of a TokenCredentials class from azure.identity.
+    :returns: Blob-updated property dict (Etag and last modified)
+    :rtype: dict[str, Any]
+    """
+    with BlobClient(blob_url, credential=credential) as client:
+        return client.upload_blob(
+            data=data,
+            blob_type=BlobType.BlockBlob,
+            overwrite=overwrite,
+            max_connections=max_connections,
+            encoding=encoding,
+            **kwargs)
+
+
+def download_blob_from_url(
+        blob_url,  # type: str
+        file_path,  # type: str
+        overwrite=False,  # type: bool
+        max_connections=1,  # type: int
+        credential=None,  # type: Any
+        **kwargs):
+    """Download the contents of a blob to a local file.
+
+    :param str blob_url:
+        The full URI to the blob. This can also include a SAS token.
+    :param str file_path:
+        The full local path, including file name, that the blob will be downloaded to.
+    :param bool overwrite:
+        Whether the local file should be overwritten if it already exists. The default value is
+        `False` - in which case a ValueError will be raised if the file already exists. If set to
+        `True`, an attempt will be made to write to the existing file.
+    :param credential:
+        The credentials with which to authenticate. This is optional if the
+        blob URL already has a SAS token or the blob is public. The value can be a SAS token string,
+        an account shared access key, or an instance of a TokenCredentials class from azure.identity.
+    :returns: Blob-updated property dict (Etag and last modified)
+    :rtype: dict[str, Any]
+    """
+    with BlobClient(blob_url, credential=credential) as client:
+        if not overwrite and os.path.isfile(file_path):
```

Comment:
This implies that we are letting the open call fail if file_path is a directory. Which also means that you will get different exception types for existing directory vs. file. Which I assume is fine.

Reply:
I think letting `os.path` raise a specific error for an invalid path might be more informative for users then us simply telling them the path already exists.